zorder/src/main/scala/org/locationtech/sfcurve/zorder/ZCurve2D.scala

```diff
@@ -49,17 +49,24 @@ class ZCurve2D(resolution: Int) extends SpaceFillingCurve2D {
     (colToMap(col), rowToMap(row))
   }
 
-  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double): Seq[(Long, Long)] = {
+  def bound(i: Long): (Double, Double, Double, Double) = {
+    val (col, row) = new Z2(i).decode
+    val x = colToMap(col)
+    val y = rowToMap(row)
+    (validateX(x - cellwidth), validateY(y-cellheight), validateX(x+cellwidth), validateY(y+cellheight))
+  }
+
+  def validateX(x: Double) = math.min(math.max(xmin, x), xmax)
+  def validateY(y: Double) = math.min(math.max(ymin, y), ymax)
+
+  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double, maxRecurse: Int = 32): Seq[(Long, Long, Boolean)] = {
     val colMin = mapToCol(xmin)
     val rowMin = mapToRow(ymax)
-    println("Input: " + xmin + " ymin: " + ymin + " xmax: " + xmax + " ymax: " + ymax)
-    val min = Z2(rowMin, colMin)
-    println("minmapToCol: " + colMin + " minmapToRow: " + rowMin + " z: " + min.z)
+    val min = Z2(colMin, rowMin)
     val colMax = mapToCol(xmax)
     val rowMax = mapToRow(ymin)
-    val max = Z2(rowMin, colMin)
```

Comment:
@anthonyccri If you have a chance, it would be immensely helpful to have a unit test which would show the bug caused by the column/row being switched.